### ISSUE TITLE
Support ESLint 6

### ISSUE THREAD

**Issue by tunnckoCore** [jest-community/jest-runner-eslint#75]
Seems like it fails when we use `overrides`

```
  ● Test suite failed to run

    ESLint configuration in /home/charlike/github/tunnckoCore/configs/.eslintrc.js is invalid:
        - Unexpected top-level property "overrides[0].extends".

      at validateConfigSchema (node_modules/jest-runner-eslint/node_modules/eslint/lib/config/conf
ig-validator.js:235:15)
```

And that's part of my eslint config

```js
  overrides: [
    {
      files: [
        '**/__tests__/**',
        '**/tests/**',
        '**/test/**',
        '**/*.{test,spec}.{js,jsx,ts,tsx,mdx}',
      ],
      extends: ['plugin:jest/recommended'],
    },
    {
      files: ['**/*.ts', '**/*.tsx'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        sourceType: 'module',
        jsx: true,
        project: 'tsconfig.json',
      },
      extends: [
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:@typescript-eslint/recommended-requiring-type-checking',

        // TODO @tunnckoCore [2200-12-12]: Consider xo-typescript
        // 'xo-typescript',

        // TODO @tunnckoCore [2200-12-12]: Do we need Prettier here too? Check how overrides works!
        'plugin:prettier/recommended',
        'prettier/@typescript-eslint',
      ],
      rules: {
        // Enforce both my and Airbnb's preference used for years in gazillion ton of code
        '@typescript-eslint/no-unused-vars': 'error',
        // Inferring is a good thing to some extent.
        '@typescript-eslint/explicit-function-return-type': 'off',
        'no-use-before-define': 'off',
        '@typescript-eslint/no-use-before-define': [
          'error',
          {
            functions: false,
            classes: true,
            variables: true,
            typedefs: true,
          },
        ],
      },
    },
    {
      // TODO @tunnckoCore [2200-12-12]: eslint errors with "uknown jsx node"
      files: ['**/*.mdx'],
      parser: 'eslint-mdx',
      plugins: ['mdx'],
      rules: {
        'mdx/no-jsx-html-comments': 'error',
        'mdx/no-unescaped-entities': 'warn',
        'mdx/no-unused-expressions': 'error',
        'mdx/remark': 'warn',
        'no-unused-expressions': 'off',
        'react/no-unescaped-entities': 'off',
      },
    },
    {
      files: ['**/*.js', '**/*.jsx', '**/*.mjs'],
      parser: 'babel-eslint',
    },
  ],

```

Or is it because eslint versions mismatch? From this error we see that the error comes from `jest-runner-eslint/node_modules/eslint` not from my node_modules (which is in the monorepo root, and `"eslint": "^6.4.0",`)?

**Comment by ljharb** [comment]
Certainly there should only be one copy of eslint in your entire tree. What does `npm ls` say? If it exits nonzero, your dep graph is invalid, and you can't expect anything to work.

**Comment by tunnckoCore** [comment]
> there should only be one copy of eslint in your entire tree

Yes, but that's the case. I use v6 in the root, and this runner has v5 as both dependency and peerDependency.

**Comment by ljharb** [comment]
Gotcha; in that case the issue is that this runner does not yet support eslint 6.

**Comment by wheeler** [comment]
@tunnckoCore Hi, I was doing some work to try to support ESLint 6 and was unable to reproduce your problem. I think you might need to update your `mdx` plugin based on this [CHANGELOG description of `eslint-mdx` version 1.5.0](https://github.com/rx-ts/eslint-mdx/blob/master/CHANGELOG.md#150-2019-09-16)

If that's not it, the error message and stack look like your config isn't working with ESLint 6 itself. I don't see anything wrong with what you posted above but maybe try/answer the following:

- Does using running using normal `eslint` instead of `jest-runner-eslint` pass?
- Can you pinpoint which config line causes the error by removing them until the runner works?
- Maybe other 'plugin' packages need to be updated?

**Comment by tunnckoCore** [comment]
> I think you might need to update your mdx plugin based on

I'm almost sure I used that version.

1. Yep.
2. I may try/check.
3. Nope. I'm always on the bleeding edge.
4. I may try these days.

> If that's not it, the error message and stack look like your config isn't working with ESLint 6 itself.

It is working. Actually, I'm not sure if Airbnb officially supports ESLint 6 yet, but it is working anyway. Both in editor and terminal, only not through the jest runner. The thing is that when I simplified (options merging) the whole core of the plugin, it seems like a bit of a mess to me (but that's a self preference) it worked so currently using a published fork under my username scope.

I'll try to put repro soon :) Thanks for the response.

**Comment by ljharb** [comment]
The Airbnb config does support eslint 6.

**Comment by tunnckoCore** [comment]
Cool. I might missed that, haha.

**Comment by SimenB** [comment]
Fixed in #77 and released in v0.7.5
